train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -1,31 +1,48 @@
 package hu.bme.mit.train.controller;
 
 import hu.bme.mit.train.interfaces.TrainController;
+import java.util.Timer;
+import java.util.TimerTask;
 
 public class TrainControllerImpl implements TrainController {
 
 	private int step = 0;
 	private int referenceSpeed = 0;
 	private int speedLimit = 0;
 	private boolean alarm = false;
+	private Timer timer = new Timer();
 
-	public void setAlarm(boolean value){
+	private static TimerTask wrap(Runnable r) {
+		return new TimerTask() {
+			@Override
+			public void run() {
+				r.run();
+			}
+		};
+	}
+
+	public TrainControllerImpl() {
+		timer.schedule(wrap(() -> {
+			followSpeed();
+		}), 200);
```

Comment:
This timer runs only once, not periodically.

Reply:
```suggestion
		}), 200, 200);
```

Reply:
(The suggestion adds the period parameter of the timer so that it runs every 200 ms.)